lib/widgets/add_expense.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../models/ModelProvider.dart';
+import '../services/api_service.dart';
+
+class AddExpense extends StatelessWidget {
+  AddExpense(this.apiService, {Key? key}) : super(key: key);
+
+  final APIService apiService;
+
+  final TextEditingController _expenseValueController = TextEditingController();
+  final TextEditingController _expenseNameController = TextEditingController();
+
+  late ExpenseCategory _selectedExpenseCategory;
+  final formGlobalKey = GlobalKey<FormState>();
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: const Text(
+        'Add Expense here',
+        textAlign: TextAlign.center,
+      ),
+      content: Form(
+        key: formGlobalKey,
+        child: Container(
+          padding: const EdgeInsets.all(16.0),
+          width: double.infinity,
+          child: SingleChildScrollView(
+            child: ListBody(children: [
+              TextFormField(
+                textAlign: TextAlign.center,
+                controller: _expenseValueController,
+                keyboardType: TextInputType.number,
+                inputFormatters: <TextInputFormatter>[
+                  FilteringTextInputFormatter.digitsOnly
+                ],
+                validator: (value) {
+                  const _validationError = 'Enter a valid expense value';
+                  if (value!.isEmpty) {
+                    return _validationError;
+                  }
+                  if (double.parse(value) <= 0) {
+                    return _validationError;
+                  }
+                  return null;
+                },
+                style: const TextStyle(fontSize: 70, color: Colors.black),
+                autofocus: true,
+                autocorrect: false,
+                decoration: const InputDecoration(hintText: "0.00"),
+                textInputAction: TextInputAction.next,
+              ),
+              const SizedBox(
+                height: 20.0,
+              ),
+              TextFormField(
+                controller: _expenseNameController,
+                autofocus: true,
+                autocorrect: false,
+                decoration: const InputDecoration(hintText: "Expense Name"),
+                textInputAction: TextInputAction.done,
+                validator: (name) {
+                  if (name!.isNotEmpty) {
```

Comment:
nit: can this be `(name?.isNotEmpty == true)` or  `(name != null && name.isNotEmpty)` for better null safety?